tests/test_main.py

```diff
@@ -14,3 +16,33 @@ def runner() -> CliRunner:
 def test_main_fails(runner: CliRunner) -> None:
     result = runner.invoke(__main__.main)
     assert result.exit_code == 2
+
+
+def test_retrieve_called_with_auth_headers(
+    request: pytest.FixtureRequest, runner: CliRunner
+) -> None:
+    if request.config.getoption("typeguard_packages"):
+        pytest.skip(
```

Comment:
I couldn't figure out a better workaround for this. It seems that running this test with the `pytest-typeguard` plugin does some funny stuff to imports which breaks my mock below. 

I believe it's coming from the fact that it does [`install_import_hook`](https://github.com/agronholm/typeguard/blob/db7771024eb2f6eba03c3954f8abe0f03bded3c2/src/typeguard/_pytest_plugin.py#L71) which makes the mock patch path no longer valid, but I couldn't find a way to fix it. The test works when running pytest without the `--typeguard-packages` option.

Reply:
Fine by me, I haven't used **typeguard** before so am unfamiliar with how it might interact w/ a running system, will defer to @philvarner on his preferred resolution.